slicedimage/io.py

```diff
@@ -182,6 +186,151 @@ def parse(self, json_doc, baseurl, backend_config):
+                    imageformat,
+                    json_doc.get(TileSetKeys.EXTRAS, None),
+                )
+
+                for tile_doc in json_doc[TileSetKeys.TILES]:
+                    relative_path_or_url = tile_doc[TileKeys.FILE]
+                    backend, name, _ = resolve_url(relative_path_or_url, baseurl, backend_config)
+
+                    tile_format_str = tile_doc.get(TileKeys.TILE_FORMAT, None)
+                    if tile_format_str:
+                        tile_format = ImageFormat[tile_format_str]
+                    else:
+                        tile_format = result.default_tile_format
+                    if tile_format is None:
+                        # Still none :(
+                        extension = os.path.splitext(name)[1].lstrip(".")
+                        tile_format = ImageFormat.find_by_extension(extension)
+                    checksum = tile_doc.get(TileKeys.SHA256, None)
+                    tile = Tile(
+                        tile_doc[TileKeys.COORDINATES],
+                        tile_doc[TileKeys.INDICES],
+                        tile_shape=Tile.format_tuple_shape_to_dict_shape(
+                            tile_doc.get(TileKeys.TILE_SHAPE, None)),
+                        sha256=checksum,
+                        extras=tile_doc.get(TileKeys.EXTRAS, None),
+                    )
+
+                    def future_maker(_source_fh_contextmanager, _tile_format):
+                        """Produces a future that reads from a file and decodes according to the
+                        specified file format."""
+                        def _actual_future():
+                            with _source_fh_contextmanager as fh:
+                                return _tile_format.reader_func(fh)
+
+                        return _actual_future
+
+                    tile.set_numpy_array_future(
+                        future_maker(
+                            backend.read_contextmanager(name, checksum_sha256=checksum),
+                            tile_format))
+                    result.add_tile(tile)
+            else:
+                raise ValueError(
+                    "JSON doc does not appear to be a collection partition or a tileset "
+                    "partition. JSON doc must contain either a {contents} field pointing to a "
+                    "tile manifest, or it must contain a {tiles} field that specifies a set of "
+                    "tiles.".format(
+                        contents=CollectionKeys.CONTENTS, tiles=TileSetKeys.TILES))
+
+            return result
+
+    class Writer(Writer):
+        def generate_partition_document(
+                self,
+                partition,
+                path,
+                pretty=False,
+                partition_path_generator=Writer.default_partition_path_generator,
+                tile_opener=Writer.default_tile_opener,
+                tile_format=ImageFormat.NUMPY,
```

Comment:
I thought we changed the default file format to TIFF?

Reply:
We changed it at a higher level: https://github.com/spacetx/starfish/pull/993